### PR TITLE
Add support for inviting through qrcode

### PR DESCRIPTION
**Pull request by devceline**
# Description
Add support by inviting through QRCode. 
- Add custom qr code scanner
- Add button to go to said scanner
- Add url for direct invite that optionally accepts public key  `/messages/invite/CAIP10ACCOUNT`
- Add `DirectInvite` component to handle invite urls 

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Examples/Screenshots (Optional)

image

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/pages/ScanQrCode/index.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    const web3inboxRegex = new RegExp(`${window.location.origin}/messages/invite/.*`, 'u')
+    if (web3inboxRegex.test(scanResult)) {
+      nav(scanResult)
+    } else {
+      nav('/messages')
+    }
+  }, [scanResult])
+
+  /*
```
Comment (boidushya) [inline src/pages/ScanQrCode/index.tsx:24]: we should remove this comment

File: src/main.tsx [hunk 2]
```diff
@@ -56,6 +60,7 @@ ReactDOM.createRoot(document.getElementById('root')!).render(
                     <Route path="/messages/chat/:peer" element={<ThreadWindow />} />
                     <Route path="/messages/new-chat" element={<NewChat />} />
                     <Route path="/messages/chat-invites" element={<ChatInvites />} />
+                    <Route path="/messages/invite/:account" element={<DirectInvite />} />
```
Comment (boidushya) [inline src/main.tsx:63]: Might just be me but I'm getting a failed to fetch toast on going to this endpoint
image

**Review by boidushya — CHANGES_REQUESTED** [review]
A couple minor issues, apart from that looks great!!

might need a bit of styling on the scan qr code page (which i could take a look at soon)

File: src/main.tsx (hunk 2 above)
Comment (devceline) [reply src/main.tsx:63]: ya it's `/` not `?`

Comment (boidushya) [reply src/main.tsx:63]: right, but this was the url i was going to `https://web3inbox-dev-hidden-git-feat-qrcode-invites-walletconnect1.vercel.app/messages/invite/0x80E50F4655b9B30Cf27D51D59C7a22fE9320eA78`
This screenshot is of my console which was throwing an error for keys.walletconnect.com

Comment (devceline) [reply src/main.tsx:63]: Was the address registered on the correct keyserver?

Comment (devceline) [reply src/main.tsx:63]: Yes I just realized, you're not using `CAIP10` address, it should be `eip155:1${address}`

Comment (boidushya) [reply src/main.tsx:63]: image
Yep that worked perfectly!
Looks sick as well 🔥

**Review by boidushya — APPROVED** [review]
LGTM  🫡

**Pull request merged**